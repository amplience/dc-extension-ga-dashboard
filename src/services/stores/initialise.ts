import {
  contentItemFilter,
  editionFilter,
  slotFilter,
} from '../../stores/ga-query-filters';
import {
  breakdown,
  contentItemIdMapping,
  editionIdMapping,
  gaApiKey,
  gaClientEmail,
  gaClientId,
  gaViewId,
  slotIdMapping,
} from '../../stores/google-analytics';
import { currencyCode, locale } from '../../stores/localization';
import type { ExtensionInstallation } from '../extension-sdk/extension-sdk.service';

export function initialiseStores(config: ExtensionInstallation): void {
  gaViewId.set(config?.googleAnalyticsViewId || '');
  gaClientId.set(config?.googleAnalyticsClientId || '');
  gaClientEmail.set(config?.googleAnalyticsClientEmail);
  gaApiKey.set(config?.googleAnalyticsKey || '');

  contentItemIdMapping.set(config?.mappings?.contentItemId);
  editionIdMapping.set(config?.mappings?.editionId);
  slotIdMapping.set(config?.mappings?.slotId);

  contentItemFilter.set(config?.filters?.contentItemFilter);
  editionFilter.set(config?.filters?.editionFilter);
  slotFilter.set(config?.filters?.slotFilter);

  breakdown.set(
    config?.breakdown || {
      dimension: 'ga:deviceCategory',
      title: 'Device Breakdown',
    }
  );

  if (config?.localization?.locale) {
    locale.set(config?.localization?.locale);
  }

  if (config?.localization?.currencyCode) {
    currencyCode.set(config?.localization?.currencyCode);
  }
}
