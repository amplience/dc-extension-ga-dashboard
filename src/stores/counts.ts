import { writable, derived } from 'svelte/store';
import type {
  SearchIndexUsersCount,
  SearchIndexSearchesCount,
  SearchIndexNoResultsRate,
} from 'dc-management-sdk-js';

export interface Counts {
  usersCount?: SearchIndexUsersCount;
  searchesCount?: SearchIndexSearchesCount;
  noResultsRate?: SearchIndexNoResultsRate;
}

export const counts = writable<Counts>({});

export const searchesPerUser = derived(counts, ($counts) => {
  return $counts?.usersCount?.count && $counts?.searchesCount?.count
    ? Math.round(
        ($counts?.searchesCount?.count / $counts?.usersCount?.count) * 1e1
      ) / 1e1
    : 0;
});
