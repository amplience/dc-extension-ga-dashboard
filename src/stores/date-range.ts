import { writable } from './persisted-store';
import { formatDateAsISOString } from '../utils/date-format';

export const DAY = 24 * 60 * 60 * 1000;

export interface DateRange {
  from: string;
  to: string;
}

const NOW = new Date().setHours(0, 0, 0, 0).valueOf();

export const dateRange = writable<DateRange>('date-range', {
  from: formatDateAsISOString(new Date(NOW - DAY * 30)),
  to: formatDateAsISOString(new Date(NOW - DAY)),
});
