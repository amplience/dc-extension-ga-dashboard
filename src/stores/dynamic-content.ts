import type { DynamicContent, Hub } from 'dc-management-sdk-js';
import { writable } from 'svelte/store';

export const client = writable<DynamicContent>(null);

export const hub = writable<Hub>(null);
