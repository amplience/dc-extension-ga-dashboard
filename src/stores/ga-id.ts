import { writable as persistedWritable } from './persisted-store';

export const gaId = persistedWritable<string>('ga-id', null);
