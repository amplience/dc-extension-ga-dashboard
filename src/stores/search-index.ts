import { writable as persistedWritable } from './persisted-store';
import type { SearchIndex } from 'dc-management-sdk-js';
import { writable } from 'svelte/store';

export const selectedIndexId = persistedWritable<string>(
  'selectedIndexId',
  null
);
export const index = writable<SearchIndex>(undefined);
export const includeReplicas = persistedWritable<boolean>(
  'includeReplicas',
  true
);
