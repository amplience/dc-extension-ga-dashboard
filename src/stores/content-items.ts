import { get, writable } from 'svelte/store';
import { client } from './dynamic-content';

function createContentItems() {
  const { subscribe, set, update } = writable([]);

  return {
    subscribe,
    fetch: async (id) => {
      const cachedContentItem = get(contentItems).find(
        (item) => item.id === id
      );
      if (cachedContentItem) {
        return cachedContentItem;
      }
      const contentItem = await get(client).contentItems.get(id);
      update((n) => [...n, ...[contentItem]]);
      return contentItem;
    },
    reset: () => set([]),
  };
}

export const contentItems = createContentItems();
