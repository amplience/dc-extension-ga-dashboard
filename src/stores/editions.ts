import { get, writable } from 'svelte/store';
import { client } from './dynamic-content';

function createEditions() {
  const { subscribe, set, update } = writable([]);

  return {
    subscribe,
    fetch: async (id) => {
      const cachedEdition = get(editions).find((item) => item.id === id);
      if (cachedEdition) {
        return cachedEdition;
      }
      const edition = await get(client).editions.get(id);
      update((n) => [...n, ...[edition]]);
      return edition;
    },
    reset: () => set([]),
  };
}

export const editions = createEditions();
