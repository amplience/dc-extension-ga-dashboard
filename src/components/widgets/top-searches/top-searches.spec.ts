import { tick } from 'svelte';
import { render } from '@testing-library/svelte';
import TopSearches from './top-searches.svelte';
import singleItemListFixture from './__fixtures__/top-search-single-item-list';
import getTopSearches from './services/get-top-searches.service';
import { index } from '../../../stores/search-index';
import type { SearchIndex } from 'dc-management-sdk-js';

jest.mock('./services/get-top-searches.service');

describe('TopSearches', () => {
  beforeEach(() => {
    index.set({ name: 'test-index-name' } as SearchIndex);
  });
  afterEach(() => {
    jest.resetAllMocks();
  });

  it('should render the TopSearches component with no data', async () => {
    (getTopSearches as jest.Mock).mockReturnValue(Promise.resolve([]));
    const { container } = render(TopSearches, {});

    await tick();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render the TopSearches component with data', async () => {
    (getTopSearches as jest.Mock).mockReturnValue(
      Promise.resolve(singleItemListFixture)
    );
    const { container } = render(TopSearches, {});

    await tick();
    expect(container.firstChild).toMatchSnapshot();
  });
});
