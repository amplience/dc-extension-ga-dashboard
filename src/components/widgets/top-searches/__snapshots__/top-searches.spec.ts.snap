// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopSearches should render the TopSearches component with data 1`] = `
<div>
  <section
    class="top-searches svelte-15uehv8"
  >
    <article
      class="svelte-pntwh0"
    >
       
      <section
        class="svelte-1kld2ef"
      >
        <div
          class="title svelte-1kld2ef"
        >
          <h2
            class="svelte-1kld2ef"
          >
            Top searches
          </h2>
           
          <div
            slot="help"
          >
            <div
              aria-expanded="false"
              class="svelte-usl6ua"
              data-testid="info-tip"
            >
              <div
                class=" svelte-zpbvvv"
                style="--icon-width: 24px; --icon-height: 24px;"
              >
                <object
                  src="ic-asset-info.svg/"
                />
              </div>
            </div>
          </div>
        </div>
         
        <div
          class="actions svelte-1kld2ef"
        >
          <div
            slot="actions"
          >
            <button
              class="download-csv-button svelte-wktmji"
              data-testid="download-csv-button"
              title="Download CSV (max 1000)"
            >
               
              <div
                class=" svelte-zpbvvv"
                style="--icon-width: 24px; --icon-height: 24px;"
              >
                <object
                  src="ic-download.svg/"
                />
              </div>
            </button>
             
            <div
              class="select top-searches-sizes-select svelte-c7jdhd"
            >
              <span>
                Show
              </span>
               
              <div
                class="
    mdc-select
    smui-select
    
    
    smui-select--standard
    
    
  "
              >
                 
                <i
                  class="mdc-select__dropdown-icon"
                />
                 
                <input
                  id="SMUI-select-1"
                  type="hidden"
                  value="5"
                />
                 
                <div
                  aria-haspopup="listbox"
                  aria-labelledby="SMUI-select-1-smui-label SMUI-select-1-smui-selected-text"
                  aria-required="false"
                  class="mdc-select__selected-text"
                  id="SMUI-select-1-smui-selected-text"
                  role="button"
                  tabindex="0"
                >
                  5
                </div>
                 
                 
                 
                <div
                  class="
    mdc-line-ripple
    
    
  "
                />
                
                 
              </div>
            </div>
          </div>
        </div>
      </section>
       
      <section
        class="widget-body svelte-4p1s5s"
      >
        <div
          class="data-table svelte-2e8um5"
        >
          <div
            class="data-table-slot 
    scrollable svelte-2e8um5"
          >
            <div
              class="data-table-head svelte-1nivie3"
            >
              <div
                class="data-table-row svelte-hige1l"
              >
                <div
                  class="data-table-cell svelte-1x8v3rk"
                  style="
  max-width:100%;
  justify-content:start;
  color:#999;"
                >
                  Search term
                   
                   
                </div>
                <div
                  class="data-table-cell svelte-1x8v3rk"
                  style="
  max-width:15%;
  justify-content:flex-end;
  color:#999;"
                >
                  Click pos.
                   
                  <div
                    class="data-table-sort null svelte-1xz2vfo"
                  />
                   
                  <div
                    class="data-table-sort-click-area svelte-1xz2vfo"
                    data-testid="data-table-sort-click-area"
                  />
                   
                </div>
                <div
                  class="data-table-cell svelte-1x8v3rk"
                  style="
  max-width:15%;
  justify-content:flex-end;
  color:#999;"
                >
                  CTR
                   
                  <div
                    class="data-table-sort null svelte-1xz2vfo"
                  />
                   
                  <div
                    class="data-table-sort-click-area svelte-1xz2vfo"
                    data-testid="data-table-sort-click-area"
                  />
                   
                </div>
                <div
                  class="data-table-cell svelte-1x8v3rk"
                  style="
  max-width:15%;
  justify-content:flex-end;
  color:#999;"
                >
                  Conversion
                   
                  <div
                    class="data-table-sort null svelte-1xz2vfo"
                  />
                   
                  <div
                    class="data-table-sort-click-area svelte-1xz2vfo"
                    data-testid="data-table-sort-click-area"
                  />
                   
                </div>
                <div
                  class="data-table-cell svelte-1x8v3rk"
                  style="
  max-width:15%;
  justify-content:flex-end;
  color:#666;"
                >
                  Count
                   
                  <div
                    class="data-table-sort desc svelte-1xz2vfo"
                  />
                   
                  <div
                    class="data-table-sort-click-area svelte-1xz2vfo"
                    data-testid="data-table-sort-click-area"
                  />
                   
                </div>
                
              </div>
            </div>
             
            <div
              class="data-table-body"
            >
              <div
                class="data-table-expandable-row  svelte-ro6k9j"
              >
                <div
                  class="row-content svelte-ro6k9j"
                  data-testid="row-content"
                >
                  <div
                    class="data-table-cell svelte-1x8v3rk"
                    style="
  max-width:100%;
  justify-content:start;
  color:#333;"
                  >
                     
                    <div
                      class="svelte-t93ore"
                    >
                      <div
                        class=" svelte-zpbvvv"
                        style="--icon-width: 24px; --icon-height: 24px;"
                      >
                        <object
                          src="ic-expandable.svg/"
                        />
                      </div>
                    </div>
                     
                    <div
                      class="query-label svelte-1ylxfod"
                    >
                      abc
                       
                      
                    </div>
                  </div>
                   
                  <div
                    class="data-table-cell svelte-1x8v3rk"
                    style="
  max-width:15%;
  justify-content:flex-end;
  color:#333;"
                  >
                    3
                  </div>
                   
                  <div
                    class="data-table-cell svelte-1x8v3rk"
                    style="
  max-width:15%;
  justify-content:flex-end;
  color:#333;"
                  >
                    25.00%
                  </div>
                   
                  <div
                    class="data-table-cell svelte-1x8v3rk"
                    style="
  max-width:15%;
  justify-content:flex-end;
  color:#333;"
                  >
                    25.00%
                  </div>
                   
                  <div
                    class="data-table-cell svelte-1x8v3rk"
                    style="
  max-width:15%;
  justify-content:flex-end;
  color:#333;"
                  >
                    <div
                      class="query-count svelte-oetuy1"
                    >
                      11
                    </div>
                  </div>
                   
                   
                </div>
                 
                <div
                  class="expandable-content svelte-ro6k9j"
                />
              </div>
              
            </div>
          </div>
        </div>
        
        
      </section>
    </article>
  </section>
</div>
`;

exports[`TopSearches should render the TopSearches component with no data 1`] = `
<div>
  <section
    class="top-searches svelte-15uehv8"
  >
    <article
      class="svelte-pntwh0"
    >
       
      <section
        class="svelte-1kld2ef"
      >
        <div
          class="title svelte-1kld2ef"
        >
          <h2
            class="svelte-1kld2ef"
          >
            Top searches
          </h2>
           
          <div
            slot="help"
          >
            <div
              aria-expanded="false"
              class="svelte-usl6ua"
              data-testid="info-tip"
            >
              <div
                class=" svelte-zpbvvv"
                style="--icon-width: 24px; --icon-height: 24px;"
              >
                <object
                  src="ic-asset-info.svg/"
                />
              </div>
            </div>
          </div>
        </div>
         
        <div
          class="actions svelte-1kld2ef"
        >
          <div
            slot="actions"
          >
            <button
              class="download-csv-button svelte-wktmji"
              data-testid="download-csv-button"
              title="Download CSV (max 1000)"
            >
               
              <div
                class=" svelte-zpbvvv"
                style="--icon-width: 24px; --icon-height: 24px;"
              >
                <object
                  src="ic-download.svg/"
                />
              </div>
            </button>
             
            <div
              class="select top-searches-sizes-select svelte-c7jdhd"
            >
              <span>
                Show
              </span>
               
              <div
                class="
    mdc-select
    smui-select
    
    
    smui-select--standard
    
    
  "
              >
                 
                <i
                  class="mdc-select__dropdown-icon"
                />
                 
                <input
                  id="SMUI-select-0"
                  type="hidden"
                  value="5"
                />
                 
                <div
                  aria-haspopup="listbox"
                  aria-labelledby="SMUI-select-0-smui-label SMUI-select-0-smui-selected-text"
                  aria-required="false"
                  class="mdc-select__selected-text"
                  id="SMUI-select-0-smui-selected-text"
                  role="button"
                  tabindex="0"
                >
                  5
                </div>
                 
                 
                 
                <div
                  class="
    mdc-line-ripple
    
    
  "
                />
                
                 
              </div>
            </div>
          </div>
        </div>
      </section>
       
      <section
        class="widget-body svelte-4p1s5s"
      >
        <div
          class="no-data-container svelte-1kq81j9"
        >
          <p>
            No data for the selected date range
          </p>
        </div>
        
        
      </section>
    </article>
  </section>
</div>
`;
